### PR TITLE
Updated keycloak access token, keycloak port, and keycloak README

### PR DESCRIPTION
**Pull request by RachalCassity**
In the **OmniauthController**, the keycloak_access_token method fetches an access token from Keycloak. Users will be able to make API calls to Argo and other clients using this token. **auth_hash['credentials']['token']** is not supposed to be used for API calls. The credential token is used for Oauth SSO.

Changed Keycloak's port to 8180 because Damien's local Argo setup uses port 8080.

### CODE REVIEW

File: app/controllers/omniauth_controller.rb [hunk 1]
```diff
@@ -20,6 +20,22 @@ def set_user
     @user = User.from_omniauth(auth_hash)
   end
 
+  def keycloak_access_token
+    return nil if auth_hash['credentials'].blank?
+
+    conn = Faraday.new
+    token_url = ENV['KEYCLOAK_TOKEN_URL'] || 'http://localhost:8180/auth/realms/Twilight/protocol/openid-connect/token'
```
Comment (ericboehs) [inline app/controllers/omniauth_controller.rb:27]:
```suggestion
    token_url = ENV.fetch('KEYCLOAK_TOKEN_URL'])
```

Let's just require the ENV var and set it in .env.development.

File: app/controllers/omniauth_controller.rb [hunk 2]
```diff
@@ -20,6 +20,22 @@ def set_user
...
+    response = conn.post(token_url) do |req|
+      req.headers =
+        {
+          'Content-Type' => 'application/x-www-form-urlencoded',
+          'Authorization' => "Bearer #{auth_hash['credentials']['token']}"
+        }
```
Comment (ericboehs) [inline app/controllers/omniauth_controller.rb:33]:
```suggestion
      req.headers = {
        'Content-Type' => 'application/x-www-form-urlencoded',
        'Authorization' => "Bearer #{auth_hash['credentials']['token']}"
      }
```

File: config/initializers/omniauth.rb [hunk 3]
```diff
@@ -3,9 +3,10 @@
 Rails.application.config.middleware.use OmniAuth::Builder do
   provider  :keycloak_openid, ENV['KEYCLOAK_CLIENT_ID'] || 'account', ENV['KEYCLOAK_CLIENT_SECRET'] || '',
             scope: 'openid profile email',
-            client_options: { site: ENV['KEYCLOAK_SITE_URL'] || 'http://localhost:8080',
+            client_options: { site: ENV['KEYCLOAK_SITE_URL'] || 'http://localhost:8180',
```
Comment (ericboehs) [inline config/initializers/omniauth.rb:6]:
```suggestion
            client_options: { site: ENV.fetch('KEYCLOAK_SITE_URL'),
```

File: app/controllers/omniauth_controller.rb (hunk 1 above)
Comment (RachalCassity) [reply app/controllers/omniauth_controller.rb:27]: Okay

**Pull request closed without merging**